src/snowflake/snowpark/_internal/data_source_utils.py

```diff
@@ -0,0 +1,279 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime
+from typing import List, Any, Tuple, Protocol, Union
+from snowflake.connector.options import pandas as pd
+from dateutil import parser
+from snowflake.snowpark.types import (
+    StringType,
+    GeographyType,
+    VariantType,
+    BinaryType,
+    GeometryType,
+    TimestampType,
+    DecimalType,
+    FloatType,
+    ShortType,
+    IntegerType,
+    BooleanType,
+    LongType,
+    TimeType,
+    ByteType,
+    DateType,
+    TimestampTimeZone,
+    StructType,
+    StructField,
+)
+
+SQL_SERVER_TYPE_TO_SNOW_TYPE = {
+    "bigint": LongType,
+    "bit": BooleanType,
+    "decimal": DecimalType,
+    "float": FloatType,
+    "int": IntegerType,
+    "money": DecimalType,
+    "real": FloatType,
+    "smallint": ShortType,
+    "smallmoney": DecimalType,
+    "tinyint": ByteType,
+    "numeric": DecimalType,
+    "date": DateType,
+    "datetime2": TimestampType,
+    "datetime": TimestampType,
+    "datetimeoffset": TimestampType,
+    "smalldatetime": TimestampType,
+    "time": TimeType,
+    "timestamp": TimestampType,
+    "char": StringType,
+    "text": StringType,
+    "varchar": StringType,
+    "nchar": StringType,
+    "ntext": StringType,
+    "nvarchar": StringType,
+    "binary": BinaryType,
+    "varbinary": BinaryType,
+    "image": BinaryType,
+    "sql_variant": VariantType,
+    "geography": GeographyType,
+    "geometry": GeometryType,
+    "uniqueidentifier": StringType,
+    "xml": StringType,
+    "sysname": StringType,
+}
+ORACLEDB_TYPE_TO_SNOW_TYPE = {
+    "bfile": BinaryType,
+    "varchar2": StringType,
+    "varchar": StringType,
+    "nvarchar2": StringType,
+    # TODO: SNOW-1922043 Investigation on handling number type in oracle db
+    "number": DecimalType,
+    "float": FloatType,
+    "long": LongType,
+    "date": DateType,
+    "intervalyeartomonth": StringType,
+    "intervaldaytosecond": StringType,
+    "json": VariantType,
+    "binary_float": FloatType,
+    "binary_double": FloatType,
+    "timestamp": TimestampType,
+    "longraw": BinaryType,
+    "raw": BinaryType,
+    "clob": StringType,
+    "nclob": StringType,
+    "blob": BinaryType,
+    "char": StringType,
+    "nchar": StringType,
+    "rowid": StringType,
+    "sys.anydata": VariantType,
```

Comment:
this data looks very special, I see there are types like ANYDATA/ANYDATASET/ANYTYPE in https://docs.oracle.com/en/database/oracle/oracle-database/19/arpls/ANYDATA-TYPE.html

is this mapped to those types?

Reply:
this is only map to SYS.ANYDATA, according to https://docs.google.com/document/d/1PLI9OnB2v2v6zJfs2rj9fLYdSymImwhX/edit,
the other two type is not defined yet